### PR TITLE
Feature/test organization endpoint

### LINKED ISSUE #2863: Add pure javascript test for /organizations endpoint
**Description**:

Currently, we don't have automated tests for our REST API.

To visualize our swagger file, 
1. open http://petstore.swagger.io/
2. insert https://apinf.io/rest/v1/swagger.json

We should have pure javascript tests running on continuous integration similarly to [apinf-fiware package](https://github.com/apinf/apinf-fiware/tree/60920bd5cdc7c0c32c6010a094b8dd5c84466de8/tests) ([travis build](https://travis-ci.org/apinf/apinf-fiware/builds/269819184?utm_source=github_status&utm_medium=notification)), that means, no chimp.js or robot tests for it.

**User Story**:

- As an APInf Platform developer,
- I need Platform /organizations endpoint automated tested running on travis-ci
- So that it's good.

### PR DESCRIPTION
**Pull request by midyan**
Closes #2863

WIP

### CODE REVIEW

**Comment by matleppa** [comment]
FYI

With PR 2876 the naming of fields in response messages have bee changed.
* status: either 'success' or 'fail'
* in failure case there is included a field named 'message' containing more detailed explanatiion
* in successful case there is included a field named 'data' containing the data
  * except in case of successful deletion, when HTTP code is 204 and no content is returned.

See more detailed information about content of response messages in swagger documents generated by REST APIs.
- http://localhost:3000/rest/v1/catalog.json
- http://localhost:3000/rest/v1/management.json

**Comment by midyan** [comment]
Thanks for the heads up, @matleppa. I'll make sure all fields are tested properly.

File: .travis.yml [hunk 1]
```diff
@@ -47,6 +47,8 @@ script:
   - travis_retry yarn test
   # Build docker image
   - ./.scripts/docker_build.sh
+  # Run Endpoint testing
```
Comment (mauriciovieira) [inline .travis.yml]: It makes more sense to be before the docker build.

**Comment by bajiat** [comment]
@midyan and @mauriciovieira Should we close this PR? Or do you plan to update and continue with it?

File: .travis.yml [hunk 2]
```diff
@@ -47,6 +47,8 @@ script:
   - travis_retry yarn test
   # Build docker image
   - ./.scripts/docker_build.sh
+  # Run Endpoint testing with latests version of NodeJS
```
Comment (mauriciovieira) [inline .travis.yml]: Run these tests before docker build. It makes no sense to run after it.

File: tests/endpoints/organization/organization.test.js [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+// Clear database before test runs
+afterAll(() => {
+  return Promise.all([
+    clearCollection('users'),
+    clearCollection('Organizations'),
+  ]);
+});
+
+describe('Endpoints for organization module', () => {
```
Comment (mauriciovieira) [inline tests/endpoints/organization/organization.test.js]: add blank line.

File: tests/endpoints/organization/organization.test.js [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+    it('should return all organizations from the collection', async () => {
+      // Define response variable
+      const { body } = await request.get(organizations.endpoint);
+
+      // Expect to be successful and to by an empty array
+      expect(body.status).toEqual('success');
+      expect(isArray(body.data)).toEqual(true);
+    });
+  });
+  describe('POST - /organizations', () => {
```
Comment (mauriciovieira) [inline tests/endpoints/organization/organization.test.js]: add blank line.

File: tests/endpoints/organization/organization.test.js [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+      expect(data.url).toEqual(newOrganization.url);
+      expect(data.contact.person).toEqual(newOrganization.contact_name);
+      expect(data.contact.phone).toEqual(newOrganization.contact_phone);
+      expect(data.contact.email).toEqual(newOrganization.contact_email);
+      expect(data.socialMedia.facebook).toEqual(newOrganization.facebook);
+      expect(data.socialMedia.instagram).toEqual(newOrganization.instagram);
+      expect(data.socialMedia.twitter).toEqual(newOrganization.twitter);
+      expect(data.socialMedia.linkedIn).toEqual(newOrganization.linkedIn);
+    });
+    it('should return 400 because of missing name paramenter', async () => {
```
Comment (mauriciovieira) [inline tests/endpoints/organization/organization.test.js]: add blank line.

**Comment by mauriciovieira** [comment]
@bajiat I think this can go to our continuous integration process.

File: .travis.yml (hunk 2 above)
Comment (midyan) [reply .travis.yml]: I need the application up running to perform the tests on top of it

**Comment by midyan** [comment]
@mauriciovieira API endpoint testing is done. I will investigate why chimp tests are failing

File: tests/endpoints/organization/organization.test.js (hunk 3 above)
Comment (midyan) [reply tests/endpoints/organization/organization.test.js]: All line breaking is linter compliant now

File: tests/endpoints/organization/organization.test.js (hunk 4 above)
Comment (midyan) [reply tests/endpoints/organization/organization.test.js]: All line breaking is linter compliant now

File: tests/endpoints/organization/organization.test.js (hunk 5 above)
Comment (midyan) [reply tests/endpoints/organization/organization.test.js]: All line breaking are linter compliant now

**Comment by midyan** [comment]
@mauriciovieira ready for review

**Comment by 55** [comment]
@mauriciovieira review and merge.

**Comment by preriasusi** [comment]
@anarva could you check

**Comment by anarva** [comment]
Started working on this!

**Comment by marla-singer** [comment]
@preriasusi What to do with this PR?

**Comment by anarva** [comment]
Let's drop this PR as obsolete.

**Comment by anarva** [comment]
Note: I will resolve the conflicts shortly and provide the information required for the review.

**Pull request closed without merging**